noxfile.py

```diff
@@ -0,0 +1,228 @@
+import pathlib
+import re
+from typing import Optional, List
+import nox
+
+nox.options.reuse_existing_virtualenvs = True
+nox.options.sessions = [
+    "black",
+    "pylint",
+    "mypy",
+    "check_yaml",
+    "check_json",
+    "check_toml",
+    "check_eof",
+    "check_trailing_space",
+    "check_lf",
+]
+
+
+@nox.session
+def black(session: nox.Session):
+    """Checks formatting linting and typing"""
+    session.install("-r", "requirements/dev-black.txt")
+    session.run("black", ".")
+
+
+@nox.session(python=False)
+def pylint(session: nox.Session):
+    """Run pylint using in default env"""
+    session.run("pip", "install", "-r", "requirements/dev-pylint.txt")
+    session.run("pylint", "sphinx_immaterial")
```

Comment:
Not sure if pylint results are python version-dependent, but if so we should also do this for all python versions.

Reply:
IDK if the python version effects the results. It might depending on certain deps used (c-ext libs come to mind). So, I just made pylint do the same that mypy does now (use all supported versions of python).